packages/makeswift-next-plugin/index.js

```diff
@@ -3,14 +3,23 @@
 /** @typedef {import('next').NextConfig} NextConfig */
 /** @typedef {{ resolveSymlinks?: boolean }} MakeswiftNextPluginOptions */
 
+const fs = require('fs')
+const path = require('path')
 const withTmInitializer = require('next-transpile-modules')
+const { promisify } = require('util')
+const glob = promisify(require('glob'))
+const swc = require('@swc/core')
 
 const NEXT_IMAGE_DOMAINS = ['s.mkswft.com']
 const NEXT_TRANSPILE_MODULES_MODULES = ['@makeswift/runtime']
 
+const defaultMakeswiftConfig = {
+  components: [],
+}
+
 /** @type {(options: MakeswiftNextPluginOptions) => (nextConfig: NextConfig) => NextConfig} */
-module.exports = ({ resolveSymlinks } = {}) => (nextConfig = {}) => {
-  /** @type {NextConfig} */
+module.exports = ({ resolveSymlinks } = {}) => async (nextConfig = {}) => {
```

Comment:
Our "enhance" function should still be sync, what should be async is what we return from this function, which would be the function-form of a Next.js config

Reply:
This is what you mean, right?
```javascript
module.exports = ({ resolveSymlinks } = {}) => (nextConfig = {}) => async () => {
  // implement
}
```

Reply:
Right, and just to be more explicit:

```ts
type MakeswiftPluginOptions = {
  resolveSymlinks?: boolean
}

function makeswiftNextPlugin(pluginOptions: MakeswiftNextPluginOptions) {
  return (nextConfig: NextConfig) => async (phase, { defaultConfig }) => {
    // apply Makeswift enhancements and custom logic here...

    /**
     * @type {import('next').NextConfig}
     */
    const nextConfig = defaultConfig

    return nextConfig
  }
}
```

Reply:
If you're not getting the `phase` argument and the second argument bag which has the `defaultConfig` then that's an easy way to know whether it's in the right place. More info here: https://nextjs.org/docs/api-reference/next.config.js/introduction

Reply:
Ok cool. I will leave it the way I have it for now, but once we move this thing to TS, I'll go with the above refactor

Reply:
(I just verified what is there based on your `phase` comment)

Reply:
> I will leave it the way I have it for now, but once we move this thing to TS, I'll go with the above refactor

Are you think we would leave the PR open until then or were you thinking about merging it?